kafka-connect-influxdb/src/main/scala/com/datamountaineer/streamreactor/connect/influx/writers/InfluxBatchPointsBuilderFn.scala

```diff
@@ -19,28 +19,136 @@ package com.datamountaineer.streamreactor.connect.influx.writers
 import java.util.concurrent.TimeUnit
 
 import com.datamountaineer.streamreactor.connect.influx.config.InfluxSettings
+import com.datamountaineer.streamreactor.connect.schemas.ConverterUtil
 import io.confluent.common.config.ConfigException
-import org.apache.kafka.connect.data.Struct
+import org.apache.kafka.connect.data.{Schema, Struct}
 import org.apache.kafka.connect.sink.SinkRecord
 import org.influxdb.InfluxDB.ConsistencyLevel
 import org.influxdb.dto.{BatchPoints, Point}
 
+import scala.collection.JavaConversions._
+import scala.util.Try
 
-object InfluxBatchPointsBuilderFn {
+
+object InfluxBatchPointsBuilderFn extends ConverterUtil {
   def apply(records: Seq[SinkRecord],
             settings: InfluxSettings): BatchPoints = {
+
+    def handleSinkRecord(record: SinkRecord): Option[Point] = {
+
+      Option(record.valueSchema()) match {
+        case None =>
+          //try to take it as string
+          record.value() match {
+            case map: java.util.Map[_, _] => buildPointFromMap(record, settings)
+            case _ => sys.error("For schemaless record only String and Map types are supported")
```

Comment:
Ant, take a look at the else branch. if you don't have a schema, and you send a json and use JsonConverter you get back a java.util.Map.  the logic there is correct. If you send an int/long/Boolean what would you insert it against in influx?